packages/apps/frontera/src/routes/finder/src/components/ContactPreviewCard/ContactPreviewCard.tsx

```diff
@@ -225,16 +223,30 @@ export const ContactPreviewCard = observer(() => {
           <Input
             size='xs'
             variant='unstyled'
-            value={role || ''}
             placeholder='Enter title'
+            value={jobRole?.jobTitle || ''}
             onFocus={(e) => e.target.select()}
             className='w-[290px] overflow-hidden text-ellipsis whitespace-nowrap'
+            onBlur={() => {
+              contact.commit();
+              set(
+                contact.value,
+                'latestOrganizationWithJobRole.jobRole.jobTitle',
+                jobRole?.jobTitle,
+              );
+            }}
```

Comment:
I'm pretty sure this commit() call before the set() will cause an unintented side-effect for the next commit() call performed on this contact.

Reply:
Commit must always be called after you finish mutating the object.